00-introduction.adoc

```diff
@@ -1,199 +1,265 @@
-toes with changes they make. The more services at play, the more complex
-everything gets, but a mature, and educated engineering department with
-a strong devops culture can eventually learn to manage this.
-
-A lot of teams who are just getting started with building services kinda
-forget about this integration part, and forget their systems are going
-to be used by a myriad of others. As companies grow, other developers
-want to start hooking into these data sources, and often the original
-developers have moved on. You end up with all these random things
-talking to other random things and the architectural diagrams (if kept
-up to date) would look like an octopus orgy.
-
-These services need to talk to each other intelligently, handle various
-types of error, have realistic timeouts, know when to retry, and most
-importantly identify themselves, so you don’t end up with a stampeding
-herd; with no idea which client is causing it.
-
-These services also need frontends, or have one mega dashboard that
-handles the UI for many of them.
-
-One of these scenarios is where you probably come in, so with the why
-covered, let’s look at how.
+production issues. If it's not as dire as that, there are at least other
+sources of confusion that'll get you spinning your wheels trying to
+figure out what to do next. 
+
+Throughout the 2000s, most of us web developers were building frontend and the
+backend in the same application. The frontend was just whatever HTML the backend
+decided to spit out. Frontend developers would design the templates (HTML + data
+tags), and the backend would decide which HTML template to show, and what values
+to shove into those tags. Go Smarty.
+
+It was a simple time, but there was a lot of difficulty and hair-pulling.
+Sharing data between the presentation layer and the the backend was bad enough,
+but sharing data between different organizations was a mess. There were various
+complicated things floating around like SOAP ("Simple Object Access Protocol"),
+but instead of diving into that there was a scary amount of nonsense like iframe
+trickery. 
+
+There was a whole lot of scraping websites and yoinking out data with regex. One
+time built something for a financial services company to provide stocks and
+shares data for the MSN Money homepage. We generated full HTML on our backend
+from some horrendous CMS, then they read the whole thing, regex replaced some
+special tags added just for them, cached it somehow, and shoved their own CSS on
+it... It was around this time I started writing about APIs. 🤔
+
+=== Rise of the API
+
+Thankfully we've mostly escaped that swamp, and the vast majority of
+organizations are building APIs for this sort of work, with API consumers
+focusing on interface building.
+
+On the desktop it's all about single-page JavaScript applications, with
+EmberJS/ReactJS offering data-binding to pull and push data from a data source.
```

Comment:
```suggestion
Angular/React/Vue/Rails offering data-binding to pull and push data from a data source.
```

Ember has [fallen out of fashion](https://trends.google.com/trends/explore?date=all&geo=US&q=%2Fm%2F0s8wr70&hl=en) by now, sadly